File: examples/javaone-2026/src/main/java/org/example/koog/java/structs/Transaction.java
package org.example.koog.java.structs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;

public record Transaction(
        @JsonProperty("id") String id,
        @JsonProperty("sender") String sender,
        @JsonProperty("recipient") String recipient,
        @JsonProperty("amount") Integer amount,
        @JsonProperty("timestamp") Instant timestamp,
        @JsonProperty("status") Status status
) {

    @JsonCreator
    public Transaction {}

    public enum Status {
        ERROR,
        SUCCESS,
        PENDING,
        REJECTED,
        DISPUTED,
        CANCELED
    }
}

File: examples/spring-io-2026/kotlin/src/main/kotlin/org/example/koog/spring/kotlin/structs/Transaction.kt
package org.example.koog.spring.kotlin.structs

import com.fasterxml.jackson.annotation.JsonCreator
import kotlin.time.Instant
import kotlinx.serialization.Serializable

@Serializable
data class Transaction @JsonCreator constructor(
    val id: String,
    val sender: String,
    val recipient: String,
    val amount: Int,
    val timestamp: Instant,
    val status: Status
) {
    enum class Status {
        ERROR,
        SUCCESS,
        PENDING,
        REJECTED,
        DISPUTED,
        CANCELED
    }
}